astropy/io/fits/_tiled_compression/tiled_compression.py

```diff
@@ -203,12 +203,13 @@ def _check_compressed_header(header):
             if not np.isreal(header[kw]) or not float(header[kw]).is_integer():
                 raise TypeError(f"{kw} should be an integer")
 
-    if "TFORM1" in header:
-        for valid in ["1PB", "1PI", "1PJ", "1QB", "1QI", "1QJ"]:
-            if header["TFORM1"].startswith(valid):
-                break
-        else:
-            raise RuntimeError(f"Invalid TFORM1: {header['TFORM1']}")
+    for suffix in range(1, 10):
```

Comment:
Not sure if it's possible to have more than 2 columns ? (`COMPRESSED_DATA` and `GZIP COMPRESSED DATA`) ?

Reply:
One can also have ZSCALE and ZZERO for instance, and I'm not sure if the standard disallows additional columns that the user might have added - but I've now changed this to use TFIELDS to know how many columns there actually are.

Reply:
Seems unlikely that there would be more columns:

> 10.1.3. Table columns
Two columns in the FITS binary table are defined below to con-
tain the compressed image tiles; the order of the columns in the
table is not significant.
[...]
COMPRESSED DATA – [variable-length; required]
GZIP COMPRESSED DATA [variable-length; optional]

But using TFIELDS seems a fine solution :+1: